crates/fuel-core/src/graphql_api/da_compression.rs

```diff
@@ -210,3 +298,78 @@ where
         anyhow::bail!("UtxoId not found in the block events");
     }
 }
+
+impl<'a, Tx, Onchain> HistoryLookup for DecompressDbTx<'a, Tx, Onchain>
+where
+    Tx: OffChainDatabaseTransaction,
+    Onchain: StorageInspect<Coins, Error = fuel_core_storage::Error>
+        + StorageInspect<Messages, Error = fuel_core_storage::Error>
+        + StorageInspect<FuelBlocks, Error = fuel_core_storage::Error>,
+{
+    fn utxo_id(
+        &self,
+        c: fuel_core_types::fuel_tx::CompressedUtxoId,
+    ) -> anyhow::Result<fuel_core_types::fuel_tx::UtxoId> {
+        if c.tx_pointer.block_height() == 0u32.into() {
+            // This is a genesis coin, which is handled differently.
+            // See CoinConfigGenerator::generate which generates the genesis coins.
+            let mut bytes = [0u8; 32];
+            let tx_index = c.tx_pointer.tx_index();
+            bytes[..std::mem::size_of_val(&tx_index)]
+                .copy_from_slice(&tx_index.to_be_bytes());
+            return Ok(fuel_core_types::fuel_tx::UtxoId::new(
+                fuel_core_types::fuel_tx::TxId::from(bytes),
+                0,
+            ));
+        }
+
+        let block_info = self
+            .onchain_db
+            .storage_as_ref::<fuel_core_storage::tables::FuelBlocks>()
+            .get(&c.tx_pointer.block_height())?
+            .ok_or(not_found!(fuel_core_storage::tables::FuelBlocks))?;
+
+        let tx_id = *block_info
+            .transactions()
+            .get(c.tx_pointer.tx_index() as usize)
+            .ok_or(anyhow::anyhow!(
+                "Transaction not found in the block: {:?}",
+                c.tx_pointer
+            ))?;
+
+        Ok(fuel_core_types::fuel_tx::UtxoId::new(tx_id, c.output_index))
+    }
+
+    fn coin(
+        &self,
+        utxo_id: fuel_core_types::fuel_tx::UtxoId,
+    ) -> anyhow::Result<fuel_core_compression::ports::CoinInfo> {
+        let coin = self
+            .onchain_db
+            .storage_as_ref::<fuel_core_storage::tables::Coins>()
+            .get(&dbg!(utxo_id))?
```

Comment:
We probably shouldn't leave the `dbg!`.

Reply:
Definitely. Why isn't the CI catching this?